euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/RightJoin.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2016-2017 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.core.client.operator;
+
+import cz.seznam.euphoria.core.annotation.audience.Audience;
+import cz.seznam.euphoria.core.client.dataset.Dataset;
+import cz.seznam.euphoria.core.client.functional.BinaryFunctor;
+import cz.seznam.euphoria.core.client.functional.UnaryFunction;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Audience(Audience.Type.CLIENT)
+public class RightJoin {
```

Comment:
How is this implementation different from `LeftJoin`? Isn't this just quite expensive syntactic sugar around the same implementation with swapped arguments?